src/OpenTK/Platform/Linux/LinuxJoystick.cs

```diff
@@ -71,29 +74,144 @@ internal sealed class LinuxJoystick : IJoystickDriver2
 
         private readonly object sync = new object();
 
-        private readonly FileSystemWatcher watcher = new FileSystemWatcher();
-
         private readonly DeviceCollection<LinuxJoystickDetails> Sticks =
             new DeviceCollection<LinuxJoystickDetails>();
 
         private bool disposed;
 
+        private const string InputDevicePath = "/proc/bus/input/devices";
+
+        private readonly Thread ProcessingThread;
+
+        private string EvdevPath
+        {
+            get
+            {
+                return Directory.Exists(JoystickPath) ? JoystickPath :
+                    Directory.Exists(JoystickPathLegacy) ? JoystickPathLegacy :
+                    String.Empty;
+            }
+        }
+
         public LinuxJoystick()
         {
-            string path =
-                Directory.Exists(JoystickPath) ? JoystickPath :
-                Directory.Exists(JoystickPathLegacy) ? JoystickPathLegacy :
-                String.Empty;
+            ProcessingThread = new Thread(ProcessEvents);
+            ProcessingThread.IsBackground = true;
+            ProcessingThread.Start();
 
-            if (!String.IsNullOrEmpty(path))
+
+            if (!String.IsNullOrEmpty(EvdevPath))
             {
-                watcher.Path = path;
+                OpenJoysticks(EvdevPath);
+            }
+        }
+
+        private MD5 Hasher = MD5.Create();
 
-                watcher.Created += JoystickAdded;
-                watcher.Deleted += JoystickRemoved;
-                watcher.EnableRaisingEvents = true;
+        private string procHash;
 
-                OpenJoysticks(path);
+        private void ProcessEvents()
+        {
+            if (!File.Exists(InputDevicePath))
+            {
+                //BSD may not have the Linux ProcFS, no use running a pointless loop
+                return;
+            }
+
+            while (!disposed)
+            {
+                /*
+                 * Rather hacky filesystemwatcher equivilant, as it doesn't work on ProcFS
+                 * As the thing is psuedo-files, we can't use the last modified date either
+                 * We have to load the thing into a stream and hash. At least it's only a few bytes.....
+                 *
+                 * Note: We could possibly override the Mono filesystem watcher backend, but this is a global variable only
+                 * and so may break user code
+                 */
+                List<int> connectedDevices = new List<int>();
+                using (StreamReader reader = new StreamReader(new FileStream(InputDevicePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite)))
+                {
+                    string newHash = BitConverter.ToString(Hasher.ComputeHash(reader.BaseStream));
+                    if (newHash == procHash)
+                    {
+                        //Marginal goto, but might as well only open the stream once
+                        goto EndCondition;
```

Comment:
Suggestion: `Thread.Sleep` at the start of the loop, and just make this a `continue` instead.